src/context/SettingsContext.jsx

```diff
@@ -22,7 +23,7 @@ const settingsReducer = (oldSettings, action) => {
 const SettingsProvider = ({ children }) => {
   const [settings, dispatch] = useReducer(
     settingsReducer,
-    JSON.parse(window.localStorage.getItem(localStorageKey)) || initialSettings
+    Object.assign({}, initialSettings, JSON.parse(window.localStorage.getItem(localStorageKey)))
```

Comment:
Good catch! That would only update when there's not settings at all (which is never the case since the theme gets set in `index.html` already—see above).